### PR TITLE
Add LiveViewTest.put_submitter/2

### LINKED ISSUE #2576: LiveViewTest.render_submit raises with value present in button
### Environment

* Elixir version (elixir -v): 1.14.3
* Phoenix version (mix deps): 1.6.16
* Phoenix LiveView version (mix deps): 0.18.18 

### Actual behavior

We are considering using a `<button name="my_thing[submit]" value="value">` to provide multiple choices to a user, with few clicks required than if we implemented a checkbox.

MDN seems to suggest that our target browsers all support the `value` attribute on buttons. When we tested this in Safari, the value came through fine in our `handle_event` callback.

Our tests (using `render_submit`) raise, however, with:

```
 ** (ArgumentError) could not find non-disabled input, select or textarea with name "my_thing[submit]"
```

### Expected behavior

I know that some sites suggest not to use button value attrs due to legacy browser support, but I would expect LiveViewTest to support buttons as a valid source of form values provided here:

https://github.com/phoenixframework/phoenix_live_view/blob/af77084fea85c7eae437f6fb19055f08c14742ab/lib/phoenix_live_view/test/client_proxy.ex#L1187

We can make the change and submit a pull request, but I thought it would be better to discuss in case there are strong opinions against this behavior. Let me know!

### PR DESCRIPTION
**Pull request by mcrumm**
The `put_submitter/2` test helper makes it possible to declare which element (or selector) will initiate the form's submit event on the client. It is meant to mimic the behavior of the [SubmitEvent: submitter property | MDN](https://developer.mozilla.org/en-US/docs/Web/API/SubmitEvent/submitter).

For example, given a LiveView form with two submit buttons:

```heex
<form id="form" phx-change="change" phx-submit="submit">
  <button id="blue" name="btn" value="blue">Blue</button>
  <button id="green" name="btn" value="green">Green</button>
</form>

<p :if={@color}>Color: <%= @color %></p>
```

...and the following `handle_event` callback:

```elixir
def handle_event("submit", %{"btn" => color}, socket) when color in ["blue", "green"] do
  {:noreply, assign(socket, :color, color)}
end
```

...then one could write tests like the following:

```elixir
import Phoenix.LiveViewTest 

test "pressing submit buttons" do
  {:ok, lv, _html} = live(...)

  form = element(lv, "#form")

  assert form
         |> put_submitter("#blue")
         |> render_submit() =~ "Color: blue"

  assert form
         |> put_submitter("#green")
         |> render_submit() =~ "Color: green"
end
```

* Related to #2490
* Closes #2576 // @sax

### CODE REVIEW

**Comment by sax** [comment]
Sweet! I appreciate how it is its own helper, as the use case is probably rare these days. Will be very useful though!

**Comment by josevalim** [comment]
@mcrumm what do you think about a `:submitter` option on `render_submit`?

**Comment by mcrumm** [comment]
@josevalim That was my first thought, but I wasn't sure the best way to include it alongside the value argument. For example, from the docs for render_submit:

```elixir
assert view
      |> form("#term", user: %{name: "hello"})
      |> render_submit(%{user: %{"hidden_field" => "example"}}) =~ "Name updated"
```

How to declare the submitter without it colliding with the form fields? Because we also already have `render_submit/3`:

```elixir
{:ok, view, html} = live(conn, "/thermo")
assert html =~ "The temp is: 30℉"
assert render_submit(view, :refresh, %{deg: 32}) =~ "The temp is: 32℉"
```

**Comment by sax** [comment]
I'll try to test out the committed implementation today.

**Comment by sax** [comment]
This seems to handle my original use case. I agree that it's better to target the element and validate that the value is valid given the HTML.

A `render_submit` implementation would be slightly more convenient for adding to my test helpers, but as written I can wrap this in a function in my test helpers to improve ease-of-use.

I don't anticipate many people using the feature, so it might be nicer to keep the functional interface to `render_submit` cleaner, unless you anticipate other options coming along.

**Comment by josevalim** [comment]
@mcrumm do we expect the last argument to always be a map? If so we can definitely distinguish the submitter option. :)

**Comment by mcrumm** [comment]
@josevalim Can you give me more info? I am not quite sure what you mean.

**Comment by josevalim** [comment]
`render_submit(view, %{user: %{"hidden_field" => "example"}}, submitter: ...)` and `render_submit(view, :refresh, %{deg: 32}, submitter: ...)`. If none of the other arguments can be lists, then it should not be ambiguous. :)

**Comment by mcrumm** [comment]
@josevalim Whoops, right now `value` could be a map or a list, so I may have introduced some ambiguity with the latest changes. :) DX-wise I think the `:submitter` option is nicer, but I am unsure about the `render_submit` pattern matches. Would you mind reviewing?

File: lib/phoenix_live_view/test/live_view_test.ex [hunk 1]
```diff
@@ -614,10 +627,28 @@ defmodule Phoenix.LiveViewTest do
       assert html =~ "The temp is: 30℉"
       assert render_submit(view, :refresh, %{deg: 32}) =~ "The temp is: 32℉"
   """
+  def render_submit(%Element{} = element, value, opts) do
```
Comment (josevalim) [inline lib/phoenix_live_view/test/live_view_test.ex:630]: The submitter is only available when there is an element? If so... maybe we should support it on the `form` function instead? Sorry for the back and forth but the API is indeed tricky (on all cases!).

Comment (mcrumm) [reply lib/phoenix_live_view/test/live_view_test.ex:630]: Yes, we only do thorough form validation with an Element. Coupling submitter to `form()` prevents reusing the same form Element with multiple submitters, but that may be the least bad choice. No worries on the back and forth, I appreciate all the consideration!

**Comment by fredwu** [comment]
Hi folks, just wondering if there's been any updates on this one, would love to see this released. :)

**Comment by josevalim** [comment]
I agree put_submitter is best. :+1:

**Comment by mcrumm** [comment]
I forgot to mention I am "out-of-office" until Saturday– if you want to move it forward I think you can _just_ `git revert 2ea8b2f` and :shipit: but if there is no rush then I will handle it when I return :)

**Review by josevalim — APPROVED** [review]

**Pull request merged**